README.md

```diff
@@ -21,68 +27,147 @@ After installing and configuring those requirements, you can download [docker-co
 
-### Install new requirements
+**Workflow** defines a structure of a business process. It uses forward moving flows referencing tasks to model it.
 
-In case you needed to add new requirements you simply need to build the docker container
-again for those to be installed and re-open shell.
+**Task** is a definition of work being part of a workflow.
+
+**Flow** is a definition on what task resp. tasks follows another.
+
+**Case** is a specific instance of a workflow.
+
+**WorkItem** is a specific instance of a task a resource has to work off.
+
+
+### Configuration
+
+Caluma is a [12factor app](https://12factor.net/) so it is configurable per environment variables.
+Different environment variable types are explained at [django-environ](https://github.com/joke2k/django-environ#supported-types)
+
+#### Common
+
+A list of configuration options which you might need to configure to get Caluma started in your environment.
+
+* `SECRET_KEY`: A secret key used for cryptography. See [more](https://docs.djangoproject.com/en/2.1/ref/settings/#std:setting-SECRET_KEY)
+* `ALLOWED_HOSTS`: A comma separated list of strings representing host/domans Caluma will be served on. See [more](https://docs.djangoproject.com/en/2.1/ref/settings/#allowed-hosts)
+* `DATABASE_NAME`: Name of database to use (default: caluma)
+* `DATABASE_USER`: username to use when connecting to the database (default: caluma)
+* `DATABASE_PASSWORD`: password to use when connecting to database
+* `DATABASE_HOST`: host to use when connecting to Postgres database (default: localhost)
+* `DATABASE_PORT`: port to use when connecting to Postgres database (default: 5432)
+* `LANGUAGE_CODE`: default language defined as fall back (default: en)
+
+#### OpenID Connect
+
+When you want to connect Caluma to your [IAM](https://en.wikipedia.org/wiki/Identity_management) supporting OpenID connect you need following settings.
+
+* `OIDC_VERIFY_ALGORITHM`: Token verification algorithm (default: HS256)
+* `OIDC_CLIENT`: Client name
+* `OIDC_JWKS_ENDPOINT`: End point of JWKS in case a asymentric algorithm is used.
+* `OIDC_SECRET_KEY`: Secret key when symmetric algorithm is used (defaults to `SECRET_KEY`)
+* `OIDC_VALIDATE_CLAIMS_OPTIONS` dict of verify signature options. See [options parameter](https://python-jose.readthedocs.io/en/latest/jwt/api.html?highlight=decode_token#jose.jwt.decode) for details.
+* `OIDC_GROUPS_CLAIM`: Name of claim to be used to represent groups (default: caluma_groups)
+
+#### Extension points
+
+Caluma is meant to be used as a service with a clean API hence it doesn't provide a Django app.
+For customization some clear extension points are defined. In case a customization is needed
+where no extension point is defined, best [open an issue](https://github.com/projectcaluma/caluma/issues/new) for discussion.
+
+##### Visibility classes
+
+Visibility classes define what user can read per node. Such can be be configured as `VISIBILITY_CLASSES`.
```

Comment:
This is the first time you mention nodes. Maybe a little explanation for people who are not familiar with GraphQL vocabulary would be helpful.

I'd also add a short conceptual explanation on how the permission system works:

> The visibility part defines what you can see at all. Anything you cannot see, you're implicitly also not allowed to modify. The visibility classes define what you see depending on your roles, permissions, etc. Building on top of this follow the permission classes (see below) that define what you can do with the data you see.

Reply:
well summarized. Integrated.